docs/specifications/4-standard-compliance-suite.md

```diff
@@ -0,0 +1,166 @@
+
+[`## Test Suite`]: #test-suite
+
+### Test Case
+Within a category title, test cases are delimited by `H4` (`#### `) titles,
+which correspond to the test case name. The test case name should be kept
+concise (dieally within 75 characters) and try to be as descriptive as possible.
+
+Immediately after the `H4` title is an english language description of the test
+case that explains the property the test is designed to validate in as much
+detail as possible. As much as possible, test case descriptions should be
+self-sufficient.
+
+After the description, a table summarizes information about the test case that
+can be useful when investigating the test:
+- **Introduced** specifies the very first *released* version of `jsii` that
+  included the test. This implies previous versions of `jsii` may not have
+  conformed to the test.
+- **Related Issues** lists any and all issues related to the test case. This
+  includes the issue that was created to track the introduction of the test, and
+  other issues such as bugs that prompted the introduction of a new compliance
+  test.
+- **Revised** lists all versions of `jsii` after the initial introduction of the
+  test case, for which the release included a revision of the test case. Trivial
+  revisions (such as correcting typos, or re-phrasing the description without
+  changing the tests case's semantic value) should not be included in the list.
+
+After the attributes table, a **TypeScript** block of code describes the
+canonical form of the test. It includes any type declaration that is used by the
+test (so the code example is self-contained). Assertions performed by the test
+should be written in the form of [`jest`] expectations.
+
+[`jest`]: https://jestjs.io/docs/en/getting-started
+
+Finally, another code block details the sequence of messages that should be
+exchanged between the *host* and `node` processes during the execution of the
+test case, such that implementations can assert conherent behavior. The dialogue
+is the sequence of JSON formatted messages, from the perspective of the *host*
+app, using the following notation:
+* Messages sent by the *host* runtime to the `node` process:
+  ```
+  > { "api": "foo" }
+  ```
+* Messages received by the *host* runtime from the `node` process:
+  ```
+  < { "result": "bar" }
+  ```
+* Comments to improve readability of the trace:
+  ```
+  # Comment continues until the end of the line
+  ```
+* Blank lines can be added to logically group trace elements
+
+> :question: is there a need to support some form of a capture mechanism to
+> provision for non-deterministric results, or non-normative elements such as
+> the exact Object IDs issued for created instances?
+
+#### Template
+
+Below is the template markdown to copy-paste when introducing a new test case in
+the compliance suite. New tests should always be added at the very end of the
+category they belong to, right after the last test in said category.
+
+````md
+### Test Category
```

Comment:
Categories might need symbolic names, will help later on in the report.

Reply:
+1, I also think test case names should be symbolic because they are often used as method names

Reply:
Could symbolic be a snake_cased form of the textual title?